packages/inter-protocol/src/proposals/replaceElectorate.js

```diff
@@ -0,0 +1,229 @@
+  /** @param {[string, Promise<Invitation>][]} addrInvitations */
+  const distributeInvitations = async addrInvitations => {
+    await Promise.all(
+      addrInvitations.map(async ([addr, invitationP]) => {
+        const [voterInvitation] = await Promise.all([invitationP]);
+        trace('Sending voting invitations to', addr);
+        await reserveThenDeposit(
+          `econ committee member ${addr}`,
+          namesByAddressAdmin,
+          addr,
+          [voterInvitation],
+        );
+      }),
+    );
+  };
+
+  await distributeInvitations(zip(values(voterAddresses), invitations));
+};
+
+const startNewEconomicCommittee = async (
+  {
+    consume: { board, chainStorage, diagnostics, zoe },
+    produce: { economicCommitteeKit, economicCommitteeCreatorFacet },
+    installation: {
+      consume: { committee },
+    },
+    instance: {
+      produce: { economicCommittee },
+    },
+  },
+  { options: { committeeName, committeeSize } },
+) => {
+  const COMMITTEES_ROOT = 'committees';
+
+  trace('startNewEconomicCommittee');
+
+  trace(`committeeName ${committeeName}`);
+  trace(`committeeSize ${committeeSize}`);
+
+  const committeesNode = await makeStorageNodeChild(
+    chainStorage,
+    COMMITTEES_ROOT,
+  );
+  const storageNode = await E(committeesNode).makeChildNode(
+    sanitizePathSegment(committeeName),
+  );
+
+  const marshaller = await E(board).getPublishingMarshaller();
+
+  trace('Starting new EC Committee Instance');
+
+  const privateArgs = {
+    storageNode,
+    marshaller,
+  };
+
+  const startResult = await E(zoe).startInstance(
+    committee,
+    {},
+    { committeeName, committeeSize },
+    privateArgs,
+    'economicCommittee',
+  );
+  const { instance, creatorFacet } = startResult;
+
+  trace('Started new EC Committee Instance Successfully');
+
+  economicCommitteeKit.resolve(
+    harden({ ...startResult, label: 'economicCommittee' }),
+  );
+
+  await E(diagnostics).savePrivateArgs(startResult.instance, privateArgs);
+
+  economicCommittee.reset();
+  economicCommittee.resolve(instance);
+
+  economicCommitteeCreatorFacet.reset();
+  economicCommitteeCreatorFacet.resolve(creatorFacet);
+
+  return creatorFacet;
+};
+
+export const replaceElectorate = async (permittedPowers, config) => {
+  const { committeeName, voterAddresses, highPrioritySendersConfig } =
+    config.options;
+
+  const economicCommitteeCreatorFacet = await startNewEconomicCommittee(
+    permittedPowers,
+    {
+      options: {
+        committeeName,
+        committeeSize: values(voterAddresses).length,
+      },
+    },
+  );
+
+  const psmKitMap = await permittedPowers.consume.psmKit;
+
+  const creatorFacets = [
+    E.get(permittedPowers.consume.reserveKit).governorCreatorFacet,
+    E.get(permittedPowers.consume.auctioneerKit).governorCreatorFacet,
+    E.get(permittedPowers.consume.vaultFactoryKit).governorCreatorFacet,
+    ...[...psmKitMap.values()].map(psmKit => psmKit.psmGovernorCreatorFacet),
```

Comment:
is there a reason to omit provisionPool and the price_feeds? They're all governed by the EC as well.

Unfortunately, with the priceFeeds, once the priceFeed coreEval goes out (see 10074), there will be old priceFeeds and new ones.It would be better if we could just upgrade the governors for the active contracts, but it's not obvious how best to distinguish them.

Reply:
> is there a reason to omit provisionPool and the price_feeds?

not really, but in the test plan for the issue here, I only called for testing one parameter of one governed contract.

Testing that they all work is explicitly in #10133 ... but you're probably right that it should be in scope here.

Reply:
> It would be better if we could just upgrade the governors for the active contracts, but it's not obvious how best to distinguish them.

I see little down-side in doing `replaceElectorate` for all of them. (again, we're not upgrading anything here... oh... but maybe we'll run into #9982. hm.)